wgsl/index.bs

```diff
@@ -989,86 +988,365 @@ Host-shareable type |S| satisfies <dfn noexport>uniform buffer layout</dfn> when
+    ptr<uniform, i32>              // Note: This is not a valid expression on its own.
+
+    // The type of a pointer value that references storage for keeping an array
+    // of 50 elements of type 'f32', using memory locations in the 'private'
+    // storage class.
+    // Here the pointee type is 'array<f32,50>'.
+    ptr<private, array<f32, 50>>   // Note: This is not a valid expression on its own.
   </xmp>
 </div>
 
-### Abstract Operations on Pointers TODO ### {#abstract-pointer-operations}
-
-A pointer value *P* supports the following operations:
-
-<table class='data'>
-  <tr><td>P.Write(V)<td>Place a value V into the referenced storage.
-               V’s type must match P’s pointee type.
-  <tr><td>P.Read()<td>An evaluation yielding the value currently in the P’s
-             referenced storage.  The result type is P's pointee type.
-  <tr><td>P.Subaccess(K)<td>Valid for pointers with a composite pointee type where
-                   *K* must evaluate to an integer between 0 and one
-                   less than the number of components in *P*’s pointee type.
-                   The subaccess evaluation yields a pointer to the storage for
-                   the K’th component within P’s referenced storage,
-                   using zero-based indexing. If P's [=storage class=] is SC, and
-                   the K'th member of P's pointee type is of type T, then
-                   the result type is `ptr<SC,T>`.
-</table>
+### Forming a pointer value ### {#forming-a-pointer-value}
+
+When evaluating an expression, a pointer value is formed in one of the
+following ways:
+
+* By using the name of a variable that is in scope.
+    The result is the pointer to the storage allocated for the variable.
+* By a <dfn noexport>composite value pointer sub-access</dfn> expression:
+    * Given a pointer to a vector, appending a single-letter vector access phrase
+        results in a pointer to the named component of the vector.
+        See [[#component-pointer-from-vector-pointer]].
+    * Given a pointer to a vector, appending an array index access phrase
+        results in a pointer to the indexed component of the vector.
+        See [[#component-pointer-from-vector-pointer]].
+    * Given a pointer to a matrix, appending an array index access phrase
+        results in a pointer to the indexed column vector of the matrix.
+        See [[#matrix-access-expr]].
+    * Given a pointer to an array, appending an array index access phrase
+        results in a pointer to the indexed element of the array.
+        See [[#array-access-expr]].
+    * Given a pointer to a structure, appending a member access phrase
+        results in a pointer to the named member of the structure.
+        See [[#struct-access-expr]].
+* By passing a pointer value to a function, where the formal parameter of the function
+     has a matching pointer type.
+     The value denoted by the formal parameter inside the called function is the same
```

Comment:
This is a bit unclear. Are we trying to say that the bit pattern representing these pointers is the same? Or something else?

Reply:
It reads as "the value of the parameter is the same as the pointer value", roughly, which is what confuses me.
Perhaps, it's the "value denoted by" semantics that I'm misreading?